testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java

```diff
@@ -377,6 +377,14 @@ public void testRangePredicate()
                 "WHERE orderkey BETWEEN 10 AND 50");
     }
 
+    @Test
+    public void testDateYearOfEraPredicate()
+    {
+        // Verify the predicate of '-1996-09-14' doesn't match '1997-09-14'
```

Comment:
It's not obvious why those two dates would compare equal. Some more details in this comment would help.